src/python.cpp

```diff
@@ -935,9 +935,30 @@ SEXP py_to_r(PyObject* x, bool convert) {
   }
 }
 
+PyObject* r_to_py(RObject x, bool convert) {
+
+  // get a static reference to the R version of r_to_py
+  Rcpp::Environment pkgEnv = Rcpp::Environment::namespace_env("reticulate");
+  Rcpp::Function r_to_py_fn = pkgEnv["r_to_py"];
+
+  // call the R version and hold the return value in a PyObjectRef (SEXP wrapper)
+  // this object will be released when the function returns
+  PyObjectRef ref(r_to_py_fn(x, convert));
```

Comment:
Just to make sure I understand the semantics, `PyObjectRef` itself is an R environment containing the associated Python wrapper object. A C finalizer is set on the Python object when this object is constructed, and that C finalizer will call `Py_DecRef()` when the parent R environment is cleaned up. To ensure that the object then lives beyond that cleanup, we need to explicitly increase the reference count (as is done below).

Reply:
Yes, that's exactly correct (note that py_to_r_cpp effectively does the same thing, incrementing the reference count before returning any Python object).

R CMD check with `gctorture()` is still running, no problems yet....